app/controllers/v1/events_controller.rb

```diff
@@ -0,0 +1,38 @@
+class V1::EventsController < ApplicationController
+  before_action :set_event, only: [:show, :update, :destroy]
+
+  def index
+    @events = Event.where(owner: @current_user)
+  end
+
+  def show
+  end
+
+  def create
+    @event = Event.new(event_params)
+    @event.save
+    render json: @event
+  end
+
+  def update
+    @event.update(event_params)
+    render json: @event
+  end
+
+  def destroy
+    @event.destroy
+    head :no_content
+  end
+
+  private
+
+  def set_event
+    @event = Event.find_by(id: params[:id], owner: @current_user)
+  rescue Mongoid::Errors::DocumentNotFound
+    head :unauthorized
+  end
```

Comment:
mmm... 
> 1) There is no event found by params[:id] --> Not found
> 2) There is not event "params[:id] for @current_user ---> Not found

I like it. It keeps a simple approach and it has sense. 

I buy your solution :+1: